cpp/src/io/json/reader_impl.cu

```diff
@@ -587,8 +587,9 @@ table_with_metadata read_json(std::vector<std::unique_ptr<datasource>>& sources,
   auto range_size        = reader_opts.get_byte_range_size();
   auto range_size_padded = reader_opts.get_byte_range_size_with_padding();
 
-  auto h_data = ingest_raw_input(
+  auto h_raw_data = ingest_raw_input(
     sources, reader_opts.get_compression(), range_offset, range_size, range_size_padded);
+  host_span<char const> h_data{reinterpret_cast<char const*>(h_raw_data.data()), h_raw_data.size()};
```

Comment:
The reason for this awkward code is that we're avoiding the copy. For JSON and CSV, this is the entire file so overhead is expected to me measurable.
Let me know if you think of a way to write this better and still avoid the copy, this is the best I came up with (i.e. localized awkwardness).

Reply:
I'm not sure if I understand. The vector is owning and has value type `uint8_t`. The host span constructed here is non-owning and has value type `char const`. I was saying that the mismatch of `uint8_t` and `char` is awkward. I don't see what would invoke a copy. This is probably a non-essential / low impact suggestion, so feel free to skip it.

Reply:
`ingest_raw_data` can't return a `vector<char>` without a copy, because decompression APIs return a vector of `uin8_t` (to be consistent with the rest of decompression stuff). JSON/CSV readers need text input, i.e. chars. So I cast when creating the span, so the rest of the reader code can deal with "text".

Reply:
Ah, that explains it. Thanks, I didn't consider that the decompression APIs were returning `uint8_t`. I don't see an immediate way to improve that. I'd resolve and move on.